add custom type and resolve modal type popup

neeta-wagento opened the pull request:
## Purpose of this pull request

<!-- REQUIRED Describe the goal and the type of changes this pull request covers. -->

This pull request (PR) is for Type of Modal is the Popup, or slide and added custom type modal 

## Related Issue
Feedback on page: /guides/v2.3/javascript-dev-guide/widgets/widget_modal.html #4524


## Affected DevDocs pages

<!-- REQUIRED List the affected pages on devdocs.magento.com (URLs). Not needed for large numbers of files. -->

- https://devdocs.magento.com/guides/v2.1/javascript-dev-guide/widgets/widget_modal.html
- https://devdocs.magento.com/guides/v2.2/javascript-dev-guide/widgets/widget_modal.html
- https://devdocs.magento.com/guides/v2.3/javascript-dev-guide/widgets/widget_modal.html

whatsnew
Clarified the modal popup types and added a 'Code sample of custom modal type' section in [Modal Widget](https://devdocs.magento.com/guides/v2.3/javascript-dev-guide/widgets/widget_modal.html).

devops-devdocs commented:
An admin must run tests on this PR before it can be merged.

dobooth commented on guides/v2.1/javascript-dev-guide/widgets/widget_modal.md:
```diff
@@ -179,7 +181,7 @@ The element that triggers the modal.
 ### `type` {#modal_type}
 
-The type of window: popup or slide.
+The type of window: popup or slide or can be the custom.
```
The type of window: `popup`, `slide`, or `custom`.

dobooth commented on guides/v2.1/javascript-dev-guide/widgets/widget_modal.md:
```diff
@@ -157,7 +159,7 @@ Sets a margin between slide modal windows.
 
 ### `responsive` {#modal_responsive}
-Turn popup modal window to slide panel on small screens. Available if the `type` option is set to `modal`.
+Turn popup modal window to slide panel on small screens. Available if the `type` option is set to `popup` or `slide` or `custom`.
```
Please put a blank line between the heading and the text.

neeta-wagento replied:
@dobooth Do I need to put a blank line after all headings?

dobooth replied:
That would be great.

dobooth commented on guides/v2.1/javascript-dev-guide/widgets/widget_modal.md:
```diff
@@ -278,8 +280,107 @@ The result is a modal and a button (_Click Here_) that opens the modal.
+## Code sample for custom type modal
+
+The following example shows how to initialize the modal widget and pass options during the initialization.
+
```
We can probably pull in a lot of indenting in this code sample.

neeta-wagento commented:
Hello @dobooth I have applied requested changes! Please review!

dobooth commented:
@neeta-wagento Looks like you need to update your base branch...

neeta-wagento commented:
Hi @dobooth I have updated my base branch, please review

dobooth commented:
Hi @neeta-wagento Thanks for the updates. Sorry but one more change. We recently stopped supporting v2.1. If you remove the v2.1 file from this PR, we can get it in.

osrecio commented:
@neeta-wagento You need some help?

neeta-wagento commented:
Hi @dobooth I have updated my base branch, please review

neeta-wagento commented:
@osrecio Thank you for heads up! I was getting issue while merging master into my branch. I have resolved it and push changes! :)

dobooth commented:
Thanks @neeta-wagento It looks like something is still not up-to-date. I cannot get the branch updated enough that it will let me merge.
Please give another try at updating. I really want to get this one in.

neeta-wagento commented:
@dobooth :  Maybe its because of  4 days ago and new changes recently merged I have updated my branch just now and added master changes please check! 
Please let me know if you are still facing any other issues !

neeta-wagento commented:
@dobooth :  Maybe its because of  4 days ago and new changes recently merged I have updated my branch just now and added master changes please check! 
Please let me know if you are still facing any other issues!

dobooth commented:
@neeta-wagento I am very sorry this PR has been so messy.
There is something amiss here.
It seems that now this PR only contains the image. And even with that, I can't get this PR into a state where it will allow me to merge. Not sure why...
If you aren't tired of this topic yet, can you please make a new, clean PR with the changes you want and we can get it in? Thanks, Don

neeta-wagento commented:
Hello @dobooth 
I am not tired yet for this PR I am closing this one and I will create new one
